tests/p2p_sync_throttle_test.py

```diff
@@ -0,0 +1,216 @@
+            startSyncThrottledBytesReceived = extractPrometheusMetric(throttledNodePortMap['9878'],
+                                                                      'block_sync_bytes_received',
+                                                                      response)
+            Print(f'Start sync throttled bytes received: {startSyncThrottledBytesReceived}')
+            break
+
+    # Throttling node was offline during block generation and once online receives blocks as fast as possible while
+    # transmitting blocks to the next node in line at the above throttle setting.
+    assert throttlingNode.waitForBlock(endLargeBlocksHeadBlock), f'wait for block {endLargeBlocksHeadBlock}  on throttled node timed out'
+    endThrottlingSync = time.time()
+    response = throttlingNode.processUrllibRequest('prometheus', 'metrics', exitOnError=True, returnType=ReturnType.raw, printReturnLimit=16).decode()
+    Print('Throttling Node End State')
+    endSyncThrottlingBytesSent = extractPrometheusMetric(throttlingNodePortMap['9879'],
+                                                         'block_sync_bytes_sent',
+                                                         response)
+    Print(f'End sync throttling bytes sent: {endSyncThrottlingBytesSent}')
+    # Throttled node is connecting to a listen port with a block sync throttle applied so it will receive
+    # blocks more slowly during syncing than an unthrottled node.
+    wasThrottled = False
+    while time.time() < endThrottlingSync + 30:
+        response = throttlingNode.processUrllibRequest('prometheus', 'metrics', exitOnError=True,
+                                                       returnType=ReturnType.raw, printReturnLimit=16).decode()
+        throttledState = extractPrometheusMetric(throttlingNodePortMap['9879'],
+                                                 'block_sync_throttling',
+                                                 response)
+        if throttledState:
+            wasThrottled = True
+            break
+    assert throttledNode.waitForBlock(endLargeBlocksHeadBlock, timeout=30), f'Wait for block {endLargeBlocksHeadBlock} on sync node timed out'
+    endThrottledSync = time.time()
+    response = throttledNode.processUrllibRequest('prometheus', 'metrics', exitOnError=True, returnType=ReturnType.raw, printReturnLimit=16).decode()
+    Print('Throttled Node End State')
+    endSyncThrottledBytesReceived = extractPrometheusMetric(throttledNodePortMap['9878'],
+                                                            'block_sync_bytes_received',
+                                                            response)
+    Print(f'End sync throttled bytes received: {endSyncThrottledBytesReceived}')
+    throttlingElapsed = endThrottlingSync - clusterStart
+    throttledElapsed = endThrottledSync - clusterStart
+    Print(f'Unthrottled sync time: {throttlingElapsed} seconds')
+    Print(f'Throttled sync time: {throttledElapsed} seconds')
```

Comment:
Can we safely check here that `throttlingElapsed` is greater than `throttledElapsed` plus some threshold?
I assume it can't be 1 sec and can't be 30 sec. So it can be like 5 or 10 seconds or so.

Reply:
In an effort to keep the test duration reasonable, we're not attempting to measure a time difference.  The test is short enough that the variance in the timing of net connections being established and the Prometheus metrics being updated (both of which happen only once every three seconds) is enough that the test is unreliable when only a time difference is measured.  As hardware improves, that will only get worse.